src/PythConnection.ts

```diff
@@ -44,7 +47,7 @@ export class PythConnection {
     }
   }
 
-  private handlePriceAccount(key: PublicKey, account: AccountInfo<Buffer>) {
+  private handlePriceAccount(key: PublicKey, account: AccountInfo<Buffer>, slot: number|null = null) {
```

Comment:
I think you forgot to pass in the slot in `handleAccount` below.

(I think it is dangerous to have the type of slot be `number | null` because it allows you to make this mistake)

Reply:
The parsing function can handle null slots and it's good to not make an extra request to get the slot when parsing right when we get the price update. I can do if we need. 
But you are right, it's a bit misleading and I will fix it.

Reply:
Fixed on this function and it gets slot from the context of the updates. it is still available in parsePriceData because I see some cases that it will cause an overhead for people to get it and they don't need it.

Reply:
One downside was that it was needed to make a request to get the current slot as onSlotChange only gives latest created slot and does not work with commitment. 

Overall, to keep things simple and be more consistent with other clients, removed the callback.